<?xml version="1.0"?>

<launch>

   <arg name="path" default="/home/aaron/Downloads/1_HuskyUrbanDataset-20201116T170146Z-001/" />
   <arg name="file_1" default="main_lidar_locus-002" />
   <arg name="file_2" default="front_lidar_locus-003" />
   <arg name="file_3" default="rear_lidar_locus-001" />
   <arg name="odom_file" default="main_lidar_odom_renamed" />

   <param name="/use_sim_time" value="true" />
   <node pkg="rosbag" type="play" name="player1" output="screen" args="--clock $(arg path)$(arg file_1).bag $(arg path)$(arg file_2).bag $(arg path)$(arg file_3).bag $(find final)/$(arg odom_file).bag" />

   <node pkg="tf2_ros" type="static_transform_publisher" name="arch_main" output="screen" args="-0.02896 0.0267886 -0.40486 -0.00095 0.00230706 -0.38147255 0.9243768 husky4/velodyne arch" />
   <node pkg="tf2_ros" type="static_transform_publisher" name="arch_front" output="screen" args="0.138 -0.01 0.285 -0.0052 0.445058 0.0 arch husky4/velodyne_front" />
   <node pkg="tf2_ros" type="static_transform_publisher" name="arch_rear" output="screen" args="-0.47 0 0.21 3.15555 0.428 0.0 arch husky4/velodyne_rear" />

   <node pkg="final" type="odom_to_tf.py" name="odom_to_tf" output="screen"/>

  <node name="octomap_main" pkg="octomap_server" type="octomap_server_node" ns="octo_main">
    <param name="frame_id" value="l1_map"/>
    <param name="filter_ground" value="false"/>
    <param name="base_frame_id" value="arch"/>
    <param name="resolution" value="0.05"/>
    <param name="height_map" value="false"/>
    <param name="color/r" value="1.0"/>
    <param name="color/g" value="0.0"/>
    <param name="color/b" value="0.0"/>
    <param name="color/a" value="0.8"/>
    <remap from="cloud_in" to="/husky4/velodyne_points" />
  </node>

  <node name="octomap_front" pkg="octomap_server" type="octomap_server_node" ns="octo_front">
    <param name="frame_id" value="l1_map"/>
    <param name="filter_ground" value="false"/>
    <param name="base_frame_id" value="arch"/>
    <param name="resolution" value="0.05"/>
    <param name="height_map" value="false"/>
    <param name="color/r" value="0.0"/>
    <param name="color/g" value="0.0"/>
    <param name="color/b" value="1.0"/>
    <param name="color/a" value="0.8"/>
    <remap from="cloud_in" to="/husky4/velodyne_front/velodyne_points" />
  </node>

  <node name="octomap_rear" pkg="octomap_server" type="octomap_server_node" ns="octo_rear">
    <param name="frame_id" value="l1_map"/>
    <param name="filter_ground" value="false"/>
    <param name="base_frame_id" value="arch"/>
    <param name="resolution" value="0.05"/>
    <param name="height_map" value="false"/>
    <param name="color/r" value="0.0"/>
    <param name="color/g" value="1.0"/>
    <param name="color/b" value="0.0"/>
    <param name="color/a" value="0.8"/>
    <remap from="cloud_in" to="/husky4/velodyne_rear/velodyne_points" />
  </node>

</launch>
